Client/Application/WindowManager.swift

```diff
@@ -18,24 +25,64 @@ struct AppWindowInfo {
 }
 
 final class WindowManagerImplementation: WindowManager {
-    private var windows: [WindowUUID: AppWindowInfo] = [:]
-    private(set) var activeWindow: WindowUUID = .defaultSingleWindowUUID /* [WIP] [FXIOS-7349] iPad multi-window. */
+    private(set) var windows: [WindowUUID: AppWindowInfo] = [:]
+    var activeWindow: WindowUUID {
+        get { return uuidForActiveWindow() }
+        set { _activeWindowUUID = newValue }
+    }
+    private var _activeWindowUUID: WindowUUID?
 
     func tabManagerDidConnectToBrowserWindow(_ tabManager: TabManager) {
         let uuid = tabManager.windowUUID
-        var windowInfo = windowInfo(for: uuid, createIfNeeded: true)
-        windowInfo?.tabManager = tabManager
-        windows[uuid] = windowInfo
+        guard var info = window(for: uuid, createIfNeeded: true) else { fatalError() }
+        info.tabManager = tabManager
+        updateWindow(info, for: uuid)
     }
 
     func tabManager(for windowUUID: WindowUUID) -> TabManager {
-        guard let tabManager = windowInfo(for: windowUUID)?.tabManager else { fatalError("No tab manager for window UUID.") }
+        guard let tabManager = window(for: windowUUID)?.tabManager else { fatalError("No tab manager for window UUID.") }
         return tabManager
     }
 
+    func windowDidClose(uuid: WindowUUID) {
+        updateWindow(nil, for: uuid)
+    }
+
     // MARK: - Internal Utilities
 
-    private func windowInfo(for windowUUID: WindowUUID, createIfNeeded: Bool = false) -> AppWindowInfo? {
+    private func updateWindow(_ info: AppWindowInfo?, for uuid: WindowUUID) {
+        assert(info != nil || windows.count > 1,
+               "Cannot remove the only active window in the app. This is a client error.")
```

Comment:
This looks like something that should send an error to Sentry, can you add a fatal error to the logger here?

Reply:
Updated to add Sentry logs; I thought this area of the app architecture might warrant a new logging category so I added `.window` to `LoggerCategory`, but I'm happy to use a different category if you prefer

Reply:
@OrlaM I just noticed this comment in `LoggerCategory`:

> `/// Do not add new categories unless discussing with the team beforehand.`

Will make sure I double-check on this before merging.